src/app/pages/wallets/my-wallets/pages-wallets-my-wallets.component.ts

```diff
@@ -37,4 +43,38 @@ export class PagesWalletsMyWalletsComponent implements OnInit {
       },
     });
   }
-}
+
+  public addNewWallet(userInputData: string) {
```

Comment:
You need to change a bit those three methods: `addNewWallet`, `updateWallet` and `openWalletModal`.
What needs to be done:
- You don't need `nowEditWallet` property at all
- You should have two public methods: `onWalletEdit` and `onWalletCreate`, they should be called when clicking on edit / create buttons in template (other good names are `handleWalletEdit` and `handleWalletCreate`)
- You should have one private method: `openWalletModal` - it should be used by `onWalletEdit` / `onWalletCreate` to open a modal. From modal you will get data for edited/created wallet OR null if modal was canceled
- If modal was canceled you do nothing, otherwise you create or update wallet using wallet's service
- You should not user property `userInputData` because it is not easy to extend it if wallet instead of only name will have more properties to edit like `description` or `initialAmount` for example